### DISCUSSION TITLE
Remote debugging custom Kafka connector

### DISCUSSION THREAD

**Discussion by ommr101** [strimzi/strimzi-kafka-operator#5282] [Q&A]
I would like to remote debug my custom connector using  Intellij.
This could be done by adding the JAVA_TOOL_OPTIONS as an env parameter with this value: -agentlib:jdwp=transport=dt_socket,address=*:5005,server=y,suspend=n.

However, I should also add port 5005 to the deployment so I can use some nodeport in order to expose the deployment's port outside of the cluster (so my IDE could attach to it).

But, I do not have the ability to change the exposed ports of the created kafka connect deployment...

Is there any workaround for that? or some config value I can use?

**Comment by scholzj** [comment]
This is not something what is supported by Strimzi. I wonder if it really makes sense to debug the connector this way instead of using local Kafka.
